### ISSUE TITLE
Add node graph building

### ISSUE THREAD

**Issue by WonderlandWar** [ruarai/CompilePal#220]
Just a small extra convenience, you'd probably need to have it load into the game to build the node graphs (building cubemaps doesn't build node graphs for some reason despite loading into the map)

**Comment by Exactol** [comment]
Not familiar with node graphs, would the nav step create it?

**Comment by WonderlandWar** [comment]
> Not familiar with node graphs, would the nav step create it?

Not sure, but I think what you could do is have the map start to build the node graphs, and to detect when the node graphs are done rebuilding, you can detect if the .ain file changes and if it does, it's likely done building the node graphs.

**Comment by SnabbSpelare** [comment]
> > Not familiar with node graphs, would the nav step create it?
> 
> Not sure, but I think what you could do is have the map start to build the node graphs, and to detect when the node graphs are done rebuilding, you can detect if the .ain file changes and if it does, it's likely done building the node graphs.

I don't think you understand. Whenever you load a map that hasn't been opened before, it will generate a Node Graph (".nav") when you're fully loaded into the map. See image: 
![image](https://github.com/user-attachments/assets/8890f93c-3da0-4e08-90fb-1b1ad2dc0c71)

But this doesn't mean it creates the AI nodes automatically—it never does. You have to manually create them by opening the console and typing "nav_generate." This cannot be done via a launch parameter, so I'm not sure what the suggestion will achieve.

This is especially true if a map hasn't been packed with a ".nav" file (which is possible for Gmod and CSGO) or if it's not included separately in the maps folder in the addon. This is a very common practice, but beginners often don't know this, based on what I've seen. Additionally, many roleplay developers don't bother adding or generating nav or AI nodes because they are useless in DarkRP and many other game modes unless there are NPCs.

By the way, you should only ever use "nav_generate" if your map actually has AI nodes. Most maps don't have any; these have to be added manually by the map developer. This cannot be automated unless by some miracle.

**Comment by SnabbSpelare** [comment]
> Just a small extra convenience, you'd probably need to have it load into the game to build the node graphs (building cubemaps doesn't build node graphs for some reason despite loading into the map)
> 
> For cubemaps, I do think it would be nice to add custom game parameters when the game starts. Like +building_cubemaps, for example.

It doesn't build nav cause your not loading into the game as a player but as a camera.
